packages/nx/src/command-line/format.ts

```diff
@@ -83,10 +83,16 @@ async function getPatterns(
     }
 
     const p = parseFiles(args);
+
     const supportedExtensions = prettier
       .getSupportInfo()
       .languages.flatMap((language) => language.extensions)
-      .filter((extension) => !!extension);
+      .filter((extension) => !!extension)
+      // Prettier supports ".swcrc" as a file instead of an extension
+      // So we add ".swcrc" as a supported extension manually
+      // which allows it to be considered for calculating "patterns"
+      .concat('.swcrc');
```

Comment:
I know we setup the json parser for the files when we generate the swcrc config, but if someone **_didnt_** use  a generator and had an swcrc file, what would happen? Is there any way for us to set the json parser here?

Reply:
That's a good point. Should we?

Reply:
If we can set it at runtime here, I'd think that'd be nicer. Looking at this code vs. the `formatFiles` utility function, we aren't programmatically calling prettier. We are just using the CLI + child_process. That may make it hard or not feasible in this case.

That said, we probably want these changes in the `formatFiles` helper too, s.t. the files are formatted if a generator changes them.

Reply:
Tried it with runtime now. A little ugly but it's better than modifying the prettier config on generation (and modifying approach only works for .prettierrc in JSON format)